website/static/js/share/utils.js

```diff
@@ -190,30 +234,253 @@ utils.removeFilter = function(vm, filter){
+    minDocCount = minDocCount || 0;
+    var ret = {'terms': {}};
+    ret.terms[field] = value;
+    ret.terms.size = 0;
+    ret.terms.exclude = 'of|and|or';
+    ret.terms.min_doc_count = minDocCount;
+    return ret;
+};
+
+/* Creates a match query */
+utils.matchQuery = function (field, value) {
+    var ret = {'match': {}};
+    ret.match[field] = value;
+    return ret;
+};
+
+/* Creates a range filter */
+utils.rangeFilter = function (fieldName, gte, lte) {
+    lte = lte || new Date().getTime();
+    gte = gte || 0;
+    var ret = {'range': {}};
+    ret.range[fieldName] = {'gte': gte, 'lte': lte};
+    return ret;
+};
+
+/* Creates a bool query */
+utils.boolQuery = function (must, mustNot, should, minimum) {
+    var ret = {
+        'bool': {
+            'must': (must || []),
+            'must_not': (mustNot || []),
+            'should': (should || [])
+        }
+    };
+    if (minimum) {
+        ret.bool.minimum_should_match = minimum;
+    }
+
+    return ret;
+};
+
+/* Creates a date histogram filter */
+utils.dateHistogramFilter = function (field, gte, lte, interval) {
+    //gte and lte in ms since epoch
+    lte = lte || new Date().getTime();
+    gte = gte || 0;
+
+    interval = interval || 'week';
+    return {
+        'date_histogram': {
+            'field': field,
+            'interval': interval,
+            'min_doc_count': 0,
+            'extended_bounds': {
+                'min': gte,
+                'max': lte
+            }
+        }
+    };
+};
+
+/* Creates a common query */
+utils.commonQuery = function (queryString, field) {
+    field = field || '_all';
+    var ret = {'common': {}};
+    ret.common[field] = {
+        query: queryString
+    };
+    return ret;
+};
+
+/* Creates a match_all query */
+utils.matchAllQuery = function () {
+    return {
+        match_all: {}
+    };
+};
+
+/* Creates an and filter */
+utils.andFilter = function (filters) {
+    return {
+        'and': filters
+    };
+};
+
+/* Creates a query filter */
+utils.queryFilter = function (query) {
+    return {
+        query: query
+    };
+};
+
+/**
+ * Parses a filter string into one of the above filters
+ *
+ * parses a filter of the form
+ *  filterName:fieldName:param1:param2...
+ *  ex: range:providerUpdatedDateTime:2015-06-05:2015-06-16
+ * @param {String} filterString A string representation of a filter dictionary
+ */
+utils.parseFilter = function (filterString) {
+        var parts = filterString.split(':');
+    var type = parts[0];
+    var field = parts[1];
+
+    // Any time you add a filter, put it here
+    switch(type) {
+        case 'range':
+            return utils.rangeFilter(field, parts[2], parts[3]);
+        case 'match':
+            return utils.queryFilter(
+                utils.matchQuery(field, parts[2])
+            );
+    }
+};
+
+utils.processStats = function (vm, data) {
+    $.map(Object.keys(data.aggregations), function (key) { //parse data and load correctly
```

Comment:
Because this function has side effects and nothing is done to with the return value of `$.map`, it'd be clearer to use a for loop or `forEach` here.